### PR TITLE
feature(list_resources): list resources per backend

### PR DESCRIPTION
**Pull request by aleksbykov**
Hydra list-resources show resources by test-id or user for all backend supported by sct. Usually the only one backend type is used. Searching on another backends could take time.
Added support to provide to hydra list-resources optional parameter backend-type which specify on which backend search resources and list them.
by default backend-type is all, and run as previously, list resources from all backends
if provide specific backend type [aws, gce, azure, eks, docker] list resources display resources only on specific backend

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
@aleksbykov that's a good direction, would love to see similar for clean resources (and passing the backend in pipeline)

File: sct.py [hunk 1]
```diff
@@ -304,8 +304,10 @@ def clean_resources(ctx, post_behavior, user, test_id, logdir, dry_run, backend)
 @click.option('--get-all-running', is_flag=True, default=False, help='All running resources')
 @sct_option('--test-id', 'test_id', help='test id to filter by')
 @click.option('--verbose', is_flag=True, default=False, help='if enable, will log progress')
+@click.option('-b', '--backend-type', type=click.Choice(["all", "aws", "gce", "eks", "docker", "azure"]),
```
Comment (fruch) [inline sct.py:307]: Those are not all the options available for backend, the list is defined in sct_config.py

File: sct.py [hunk 2]
```diff
@@ -304,8 +304,10 @@ def clean_resources(ctx, post_behavior, user, test_id, logdir, dry_run, backend)
...
+              default="all", help="use specific backend")
 @click.pass_context
-def list_resources(ctx, user, test_id, get_all, get_all_running, verbose):
+def list_resources(ctx, user, test_id, get_all, get_all_running, verbose, backend_type):
```
Comment (fruch) [inline sct.py:310]: Maybe we should consider add a call to this after clean resources, so in cases of keep resources it would show them in the log ?

**Comment by fruch** [comment]
> @aleksbykov that's a good direction, would love to see similar for clean resources (and passing the backend in pipeline)

@vponomaryov already did similar work for clean resources

@aleksbykov are you picking this up ?

File: sct.py (hunk 1 above)
Comment (fruch) [reply sct.py]: fixed

File: sct.py (hunk 2 above)
Comment (fruch) [reply sct.py]: was added to clean resources

**Review by soyacz — DISMISSED** [review]
assumming it works, LGTM

**Pull request merged**